sklearn/utils/validation.py

```diff
@@ -847,7 +850,10 @@ def check_array(
     # When all dataframe columns are sparse, convert to a sparse array
     if hasattr(array, "sparse") and array.ndim > 1:
         with suppress(ImportError):
-            from pandas.api.types import is_sparse
+            from pandas import SparseDtype  # noqa: F811
```

Comment:
flake8 gives an error:
```
sklearn/utils/validation.py:853:13: F811 redefinition of unused 'SparseDtype' from line 772
```

It may well be that the code above already defines `SparseDtype` in a `if` clause, I feel this is quite brittle to rely on this and I would rather silence flake8 in this case.